h/queue.py

```diff
@@ -11,6 +13,8 @@ def __init__(self, namespace, *args, **kwargs):
     def publish(self, topic, data):
         if self.namespace is not None:
             topic = '{0}-{1}'.format(self.namespace, topic)
+        if not isinstance(data, str):
```

Comment:
I started looking into adding JSON deserialization to the reader yesterday. I think it makes sense but is a large enough change that it would be better in a separate PR